### ISSUE TITLE
Bug: [no-array-constructor] Does not detect Array in a sequence expression

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#19494]
### Environment

Node version: 22.14.0
npm version: 11.2.0
Local ESLint version: 9.22.0
Global ESLint version: n/a
Operating System: n/a

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
  {
    rules: {
      "no-array-constructor": "error",
    },
  },
];
```
</details>

```js
new (0, Array)();
```

### What did you expect to happen?

Sequence expressions return their last rightmost element, so this should be detected as a usage of `Array()`.

### What actually happened?

No report.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tYXJyYXktY29uc3RydWN0b3I6IFwiZXJyb3JcIiovXG5cbkFycmF5KCk7XG5cbm5ldyAoMCwgQXJyYXkpKCk7Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

Seen while reviewing #19493. This bug -if it is one- is also present in [`@typescript-eslint/no-array-constructor`](https://github.com/typescript-eslint.io/rules/no-array-constructor).

**Comment by nzakas** [comment]
I'm not sure I'd consider this a bug. The ultimate goal of the rule is to encourage the use of array literal notation instead of the `Array` constructor. In this case, there's no way to replace the call to `Array() with an array literal, so I don't think we should make a change.

@eslint/eslint-tsc thoughts?

**Comment by mdjermanovic** [comment]
I think it can be replaced with `(0, [])` or just `[]`, but this example seems like a very edge case. Why would `new (0, Array)()` be used in the code instead of just `new Array()`?

**Comment by fasttime** [comment]
There are many ways to get the `Array` constructor out of an expression, and sequence expressions are just one of them. I'm not opposed to updating the rule to account for sequence expressions if someone wants to make that change, but I don't think we can ever cover all possible cases.

**Comment by fasttime** [comment]
> I'd say this use case probably falls into the last 5% of use cases, and as I already mentioned, I don't think this is actually a case where you want to say "use an array literal instead" because the intent is not clear.

I agree. This isn't a bug, it would be an enhancement to the rule, but it would cover a usage that seems uncommon and that could indicate a different problem than what the rule seeks to address.

**Comment by mdjermanovic** [comment]
Looks like we agree this is not a bug, and we're not in favor of updating the rule to cover this uncommon pattern, so closing.
